docs/guides/intro-to-tests.md

```diff
@@ -0,0 +1,154 @@
+
+The points above form a **specification** of how the function `createWordCount` should behave - in other words, the specification is a **description of how the function should work**. Once we have a specification for how the function should work we can create a number of cases showing how we expect the function `createWordCount` to behave when it is called with certain inputs.
+
+### Case 1 🖊️
+
+```js
+const input = "hello";
+const result = createWordCount(input);
+```
+
+In example 1, `input` stores a string `"hello"` which we pass as an argument to `createWordCount`.
+We then store the return value of `createWordCount` in a variable called `output`.
+In this case, we would expect the `output` to be:
+
+```js
+{ "hello" : 1 }
+```
+
+Remember that `createWordCount` needs to return an object with a count of each word in the string. In the case where the input is `"hello"` we only have a single word string and therefore a count of 1 for `"hello"`.
+
+### Case 2 🖊️
+
+```js
+const input = "north by north west";
+const result = createWordCount(input);
+```
+
+In example 2, `input` stores a string `"north by north west"` which we pass as an argument to `createWordCount`.
+We then store the return value of `createWordCount` in a variable called `output`.
+In this case, we would expect the `output` to be:
+
+```js
+{
+    "north" : 2,
+    "by" : 1,
+    "west" : 1
+}
+```
+
+## 🧪 What is a test ?
+
+We can now define what a test is in software engineering.
+
+> 🔑 A **test** is any piece of code that **checks a certain unit of code** ( typically a function ) **behaves in a particular way**.
+
+For example, we could write a test to check that `createWordCount` behaves in a particular way when passed a string containing a single word. For this example, we're using a popular JavaScript testing framework called [Jest](https://jestjs.io/docs/getting-started) to write the test code. We'll examine the example below written using Jest, and then break down the different parts of the test code to understand how it all works.
+
+```js
+test("will count the number of words for a single word string", function () {
+  const input = "hello";
+  const actualResult = createWordCount(input);
+  const expectedResult = {
+    hello: 1,
+  };
+  expect(actualResult).toEqual(expectedResult);
+});
+```
+
+Let's break down the different parts of this test code.
+
+## Anatomy of a test
+
+#### 1. Test description
+
+```js
+test("will count the number of words in a single word string", function () {});
+```
+
+We use a function in Jest called `test` to write a test description - `"will count the number of words for a single word string"` which describes the behaviour we expect in this particular test. There is also a function `function () {}` in which we will write our test code.
+
+#### 2. Test setup
+
+```js
+const input = "hello";
+const actualResult = createWordCount(input);
+const expectedResult = {
+  hello: 1,
+};
```

Comment:
What do you think about moving `expectedResult` down to the assertion section below? This way, we roughly follow the Arrange-Act-Assert pattern (although we're not really doing much "arranging").